packages/internet-header/src/services/search/coveo.service.ts

```diff
@@ -15,25 +17,36 @@ export const getCoveoSuggestions = async (query: string): Promise<CoveoCompletio
   const config = state.localizedConfig.header.search;
   const { token, organisation } = coveo.environment[state.environment];
   const url = `${coveo.url}?q=${query}&locale=${state.currentLanguage}&searchHub=${config.searchHubName}&pipeline=${config.searchPipelineName}&organizationId=${organisation}`;
-  let coveoCompletions: CoveoCompletion[] | undefined;
 
-  try {
-    const coveoResponse = await fetch(url, {
+  if (suggestionsController) suggestionsController.abort();
+  suggestionsController = new AbortController();
+
+  return new Promise((resolve, reject) => {
+    fetch(url, {
+      signal: suggestionsController.signal,
       method: 'GET',
       headers: {
         Authorization: `Bearer ${token}`,
       },
-    });
-    const coveoJSON = (await coveoResponse.json()) as CoveoResponse;
-    coveoCompletions = coveoJSON.completions.map(completion => ({
-      ...completion,
-      redirectUrl: getSearchPageUrl(completion.expression),
-    }));
-  } catch (error) {
-    console.error(
-      'Connection to coveo failed. Did you add "*.coveo.com" to your connect-src content security policy and tried turning off your adblocker?',
-    );
-  }
-
-  return coveoCompletions === undefined ? [] : coveoCompletions;
+    })
+      .then(response => response.json())
+      .then((coveoJSON: CoveoResponse) => {
+        resolve(
+          coveoJSON.completions.map(completion => ({
+            ...completion,
+            redirectUrl: getSearchPageUrl(completion.expression),
+          })),
+        );
+      })
+      .catch(error => {
+        if (error.name === 'AbortError') {
+          console.warn('Coveo suggestions request aborted.');
+        } else {
+          console.error(
+            `Fetching coveo suggestions failed, ${error}\nDid you add "*.coveo.com" to your connect-src content security policy and tried turning off your adblocker?`,
+          );
+          reject(error);
+        }
```

Comment:
Would it be a good idea to reject anyway, no matter what error happened?

```suggestion
        }
        
        reject(error);
```

Reply:
Failing to resolve or reject a promise just fails to ever change the state from "pending" to anything else. This doesn't cause any fundamental problem in Javascript because a promise is just a regular Javascript object. The promise will still get garbage collected (even if still pending) if no code keeps a reference to the promise.

The real consequence here is what does that mean to the consumer of the promise if its state is never changed? Any `.then()` or `.catch()` listeners for resolve or reject transitions will never get called. Most code that uses promises expects them to resolve or reject at some point in the future. If they don't, then that code generally never gets to finish its work.

Or if code is using `await` on the promise instead of `.then()`, then that function will just remain suspended forever on that await. The rest of the function will be dead code and will never execute.